File: shuttle_tablet/lib/features/clients/domain/entities/invoice_line_item.dart
class InvoiceLineItem {
  final String id;
  final String? tripId;
  final String lineType; // PassengerService | Cargo
  final String description;
  final double unitRate;
  final double quantity;
  final double lineTotal;
  final int sortOrder;

  const InvoiceLineItem({
    required this.id,
    this.tripId,
    required this.lineType,
    required this.description,
    required this.unitRate,
    required this.quantity,
    required this.lineTotal,
    required this.sortOrder,
  });
}

File: shuttle-api/src/ShuttleApi.Domain/Billing/InvoiceLineItem.cs
using ShuttleApi.Domain.Common;

namespace ShuttleApi.Domain.Billing;

public sealed class InvoiceLineItem : Entity<Guid>
{
    private InvoiceLineItem() { }

    public Guid InvoiceId { get; private set; }
    public Guid? TripId { get; private set; }
    public InvoiceLineType LineType { get; private set; }
    public string Description { get; private set; } = default!;
    public decimal UnitRate { get; private set; }
    public decimal Quantity { get; private set; }
    public decimal LineTotal { get; private set; }
    public int SortOrder { get; private set; }

    public static InvoiceLineItem Create(
        Guid invoiceId,
        Guid? tripId,
        InvoiceLineType lineType,
        string description,
        decimal unitRate,
        decimal quantity,
        int sortOrder)
    {
        Guard.AgainstNullOrEmpty(description, nameof(description));

        return new InvoiceLineItem
        {
            Id = Guid.NewGuid(),
            InvoiceId = invoiceId,
            TripId = tripId,
            LineType = lineType,
            Description = description,
            UnitRate = unitRate,
            Quantity = quantity,
            LineTotal = unitRate * quantity,
            SortOrder = sortOrder
        };
    }

    public void Update(string description, decimal unitRate, decimal quantity)
    {
        Guard.AgainstNullOrEmpty(description, nameof(description));
        Description = description;
        UnitRate = unitRate;
        Quantity = quantity;
        LineTotal = unitRate * quantity;
    }
}
